editor/scene_tree_dock.cpp

```diff
@@ -523,6 +523,76 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 			if (dupsingle)
 				editor->push_item(dupsingle);
 
+		} break;
+		case TOOL_COPY: {
+
+			if (!edited_scene)
+				break;
+
+			if (editor_selection->is_selected(edited_scene)) {
+				current_option = -1;
+				accept->get_ok()->set_text(TTR("I see..."));
+				accept->set_text(TTR("This operation can't be done on the tree root."));
+				accept->popup_centered_minsize();
+				break;
+			}
+
+			if (!_validate_no_foreign())
+				break;
+
+			List<Node *> selection = editor_selection->get_selected_node_list();
+			if (selection.size() == 0)
+				break;
+
+			if (!copied_nodes.empty())
+				copied_nodes.clear();
+
+			copied_nodes = selection;
+
+		} break;
+		case TOOL_PASTE: {
+
+			if (!edited_scene)
+				break;
+			if (copied_nodes.empty()) {
+				current_option = -1;
+				accept->get_ok()->set_text(TTR("I see..."));
+				accept->set_text(TTR("Nothing in copy. Sorry Dude. Copy Something First."));
```

Comment:
Normal sentence case should be used in dialogs (don't capitalize every word). Also, I would advise making it simpler and dropping the *"Sorry Dude"*; something like *"Clipboard is empty, nothing to copy."* should be enough.